apps/condo/domains/billing/schema/ValidateQRCodeService.js

```diff
@@ -111,15 +140,72 @@ const ValidateQRCodeService = new GQLCustomSchema('ValidateQRCodeService', {
+                    throw new GQLError(ERRORS.RECEIPT_ALREADY_PAID, context)
+                }
+
+                const acquiringContexts = await find('AcquiringIntegrationContext', {
                     organization: { id_in: organizations.map((org) => org.id), deletedAt: null },
                     status: CONTEXT_FINISHED_STATUS,
                     deletedAt: null,
                 })
 
                 if (isEmpty(acquiringContexts)) throw new GQLError(ERRORS.NO_ACQUIRING_CONTEXT, context)
 
-                return { qrCodeFields }
+                /** @type {BillingReceipt} */
+                let foundReceipt
+                let amount
+                const setDataFromReceipt = (receipt) => {
+                    foundReceipt = receipt
+                    amount = foundReceipt.toPay
+                }
+
+                /** @type {TCompareQRResolvers} */
+                const resolvers = {
+                    onNoReceipt: () => {
+                        amount = qrCodeAmount
+                    },
+                    onReceiptPeriodEqualsQrCodePeriod: setDataFromReceipt,
+                    onReceiptPeriodNewerThanQrCodePeriod: setDataFromReceipt,
+                    onReceiptPeriodOlderThanQrCodePeriod: (lastBillingReceipt) => {
+                        foundReceipt = lastBillingReceipt
+                        amount = qrCodeAmount
+                    },
+                }
+
+                await compareQRCodeWithLastReceipt(qrCodeFields, resolvers)
+
+                // Calculate fees
+                const acquiringContext = acquiringContexts[0]
+                const acquiringIntegration = await getById('AcquiringIntegration', acquiringContext.integration)
+                let acquiringDistributionSchema = get(acquiringIntegration, 'explicitFeeDistributionSchema', [])
+                let contextDistributionSchema = get(acquiringContext, 'implicitFeeDistributionSchema', [])
+                if (!Array.isArray(acquiringDistributionSchema)) {
+                    acquiringDistributionSchema = []
+                }
+                if (!Array.isArray(contextDistributionSchema)) {
+                    contextDistributionSchema = []
+                }
+                const formula = compactDistributionSettings(acquiringDistributionSchema.concat(contextDistributionSchema))
+                const feeCalculator = new FeeDistribution(formula)
+
+                const { type, explicitFee = '0' } = feeCalculator.calculate(amount)
+
+                const explicitFees = type === 'service' ? {
+                    explicitServiceCharge: String(explicitFee),
+                    explicitFee: '0',
+                } : {
+                    explicitServiceCharge: '0',
+                    explicitFee: String(explicitFee),
+                }
+
+                return {
+                    qrCodeFields,
+                    lastReceiptData: foundReceipt ? pick(foundReceipt, ['id', 'period', 'toPay']) : null,
```

Comment:
why do you need to filter result object by these properties?

Reply:
I'd like to return only these fields